libparsec/crates/protocol/schema/authenticated_cmds/invite_greeter_step.json5

```diff
@@ -0,0 +1,216 @@
+                    "type": "GreeterStep"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok",
+                "fields": [
+                    {
+                        "name": "claimer_step",
+                        "type": "ClaimerStep"
+                    }
+                ]
+            },
+            {
+                // The claimer has not submitted its step yet
+                "status": "not_ready"
+            },
+            {
+                // The invitation has already been completed
+                "status": "invitation_completed"
+            },
+            {
+                // The invitation has been cancelled
+                "status": "invitation_cancelled"
+            },
+            {
+                // The author is no longer part of the allowed greeters for this invitation
+                // An example of valid case for this error happens for a user invitation,
+                // if the profile of the author changes from ADMIN to NORMAL after
+                // `invite_greeter_start_greeting_attempt` was called by the claimer
+                "status": "author_not_allowed"
+            },
+            {
+                // The greeting attempt id doesn't correspond to any existing attempt
+                "status": "greeting_attempt_not_found"
+            },
+            {
+                // The author did not join the greeting attempt
+                // This should not happen, since joining is required to get the greeting attempt ID
+                "status": "greeting_attempt_not_joined"
+            },
+            {
+                // The greeting attempt has been cancelled
+                "status": "greeting_attempt_cancelled",
+                "fields": [
+                    {
+                        "name": "origin",
+                        "type": "GreeterOrClaimer"
+                    },
+                    {
+                        "name": "timestamp",
+                        "type": "DateTime"
+                    },
+                    {
+                        "name": "reason",
+                        "type": "CancelledGreetingAttemptReason"
+                    }
+                ]
+            },
+            {
+                // The submitted step is too advanced
+                // Every step before must have been completed by both peers
+                "status": "step_too_advanced"
+            },
+            {
+                // The submitted step somehow changed during polling
+                "status": "step_mismatch"
+            }
+        ],
+        "nested_types": [
+            {
+                // GreeterStep should be identical to the one in invite_claimer_step.json5
+                "name": "GreeterStep",
+                "discriminant_field": "step",
+                "variants": [
+                    {
+                        "name": "WaitPeer",
+                        "discriminant_value": "0",
```

Comment:
Use proper value instead of fake number ? the discriminant value is always a string, so better to use a value that is more human readable :thinking:

Reply:
@touilleMan I don't have a strong opinion about this, what do you think?

As far as I understand, the discriminant value is only used by serde to discriminate when loading bytes, do does it need to be human readable? Plus, it documents the order of the steps so I found numbering them useful.

Reply:
e53026bc1b0f71e1c57ed01f31d961201056d4d3

Reply:
Fixed to have the discriminant value match the name, and include the step number in the name